docs/conf.py

```diff
@@ -298,3 +306,28 @@ def _load_theme(
 
 # If true, show URL addresses after external links.
 # man_show_urls = False
+
+
+def generate_tmt_docs(app: Sphinx) -> None:
+    # Migrate the Makefile commands to sphinx native
+    project_root = Path(app.srcdir).parent
+    conf_dir = Path(app.confdir)
+
+    (conf_dir / "spec").mkdir(exist_ok=True)
+    (conf_dir / "stories").mkdir(exist_ok=True)
+    # TODO: These can be simplified by using modules
+    # generate-lint-checks
+    subprocess.run([f"{conf_dir / 'scripts' / 'generate-lint-checks.py'}",
+                   "lint-checks.rst.j2", "spec/lint.rst"], cwd=conf_dir)
+    # generate-stories
+    subprocess.run([f"{conf_dir / 'scripts' / 'generate-stories.py'}", "story.rst.j2"],
+                   cwd=conf_dir)
+    # TODO: This can use sphinx.ext.autodoc directly
+    # generate-autodocs
+    subprocess.run(["sphinx-apidoc", "--force", "--implicit-namespaces", "--no-toc",
+                    "-o", f"{conf_dir / 'autodocs'}", "tmt"],
+                   cwd=project_root)
```

Comment:
I would call this migration - commands still exist in the Makefile. Now we have them twice :)

I'm not very fond of the form of this move, I was pretty happy when I moved exactly this kind of code from `conf.py` into `Makefile` two weeks ago :) `make` is better suited for calling scripts and taking actions based on the existence of files or their timestamps, which is exactly what I wanted to take advantage of, to not regenerate docs from sources when it's not necessary. There will be more scripts like this, e.g. a script to generate plugin docs from their docs strings, and so on.

I'd prefer leaving tasks like this to `make`, and there's no reason why this function couldn't use the `Makefile` and re-use the existing targets rather than reverting them into their previous Python state. Can you try whether running `make -C docs generate` from this function would give you the desired results?

Reply:
Hmm, fair point, but shouldn't all of the fmf spec files be marked as a dependency for this?

To play the devil's advocate, if we take use the `conf.py` builder hooks, we can work on exporting these hooks as sphinx language domains, e.g. as a `.. fmf:autoplan: /path/to/tmt/tree` (like `.. py:automodule`). Sphinx itself should be able to keep track of file dependencies with that.

But otherwise either approach is fine with me as long as there is consistency and we've considered what each tool has to offer. Indeed the use-case you've described seems very useful, but shouldn't it go further to pull in the source dependencies as well than?

Reply:
> Hmm, fair point, but shouldn't all of the fmf spec files be marked as a dependency for this?

I guess that depends... E.g. for lint checks, `make` would generate a ReST file from Python sources, so no fmf files are involved, just sources. I suppose there will be multiple shades to this... `generate-stories` does collect various fmf files into a single ReST file, yes, but other `generate-*` targets won't, so...

I for one prefer a rather incremental approach :) A custom Sphinx domain sounds very interesting, but I suppose that would be no trivial task, right? For me, the primary concern was reducing the duplication, making the code the source of truth, and `make` & `jinja2` served that purpose perfectly & quickly.